subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClasspathUtil.java

```diff
@@ -133,7 +133,7 @@ public static File getClasspathForResource(URL resource, String name) {
                 String schemeSpecificPart = location.getRawSchemeSpecificPart();
                 int pos = schemeSpecificPart.indexOf("!");
                 if (pos > 0) {
-                    assert schemeSpecificPart.substring(pos + 1).equals("/" + name);
+                    assert schemeSpecificPart.substring(pos + 1).endsWith(name);
```

Comment:
I would keep the assertion consistent with the one on line 130 and check `.endsWith("/" + name)`

Reply:
Fixe.